com.unity.render-pipelines.high-definition/Editor/Lighting/LightUnit/LightUnitSlider.cs

```diff
@@ -202,19 +240,141 @@ protected override float GetPositionOnSlider(float value)
             return ValueToSlider(value);
         }
 
-        protected override void DoSlider(Rect rect, SerializedProperty value, Vector2 range)
+        protected override void DoSlider(Rect rect, SerializedProperty value, Vector2 sliderRange)
         {
             value.floatValue = ExponentialSlider(rect, value.floatValue);
         }
 
         float ExponentialSlider(Rect rect, float value)
         {
-            var internalValue = GUI.HorizontalSlider(rect, ValueToSlider(value), 0f, 1f, GUI.skin.horizontalSlider, GUI.skin.horizontalSliderThumb);
-
+            var internalValue = GUI.HorizontalSlider(rect, ValueToSlider(value), 0f, 1f);
             return SliderToValue(internalValue);
         }
     }
 
+    /// <summary>
+    /// Formats the provided descriptor into a piece-wise linear slider with contextual slider markers, tooltips, and icons.
+    /// </summary>
+    class PiecewiseLightUnitSlider : LightUnitSlider
+    {
+        struct Piece
+        {
+            public Vector2 domain;
+            public Vector2 range;
+            public Func<float, float> transform;
```

Comment:
As the algorithm does not change between transform and inverse transform, the use of `Func` is not appropriate: it is a reference function, possibly a null pointer.

You can store the `m` and `b` variables for direct and inverse transform and have a method to apply the algorithm instead.

If you are looking for reusable code you can have:
```
static float DoTransform(in float x, in float m, in float b) => (m * x) + b;
float Transform(in float x) => DoTransform(x, directM, directB);
float InverseTransform(in float x) => DoTransform(x, inverseM, inverseB);
```